### PR TITLE
cpu/stm32: add support for STM32C0 and NUCLEO-C031C6 

### LINKED ISSUE #19210: support for new STM32C0 MCU and eval boards
#### Description

STM releases a new-brand STM32C0 MCU  and a new eval boards :  nucleo-c031c6, stm32c0116-dk, stm32c0316-dk

The STM32C0 MCU targets 8 bits and 16 bits applications according the STM marketing newsletter.

### Useful links

* https://www.st.com/content/st_com/en/products/evaluation-tools/product-evaluation-tools/mcu-mpu-eval-tools/stm32-mcu-mpu-eval-tools/stm32-nucleo-boards/nucleo-c031c6.html
* https://www.st.com/content/st_com/en/products/evaluation-tools/product-evaluation-tools/mcu-mpu-eval-tools/stm32-mcu-mpu-eval-tools/stm32-discovery-kits/stm32c0116-dk.html
* https://www.st.com/content/st_com/en/products/evaluation-tools/product-evaluation-tools/mcu-mpu-eval-tools/stm32-mcu-mpu-eval-tools/stm32-discovery-kits/stm32c0316-dk.html

### PR DESCRIPTION
**Pull request by jparker324**
Cleaned-up commit history of PR #20267 

Support added for ST Micro's board `NUCLEO-C031C6`, it's cpu `STM32C031C6`, and the rest of the cpu family `STM32C0XX` (as `CPU_FAM_STM32C0`).  This currently includes `STM32C011` and `STM32C031`, but code for identification has also been  added for the `STM32C071` and `STM32C091` lines due out later this year.

Code was adapted from `CPU_FAM_STM32G0` where applicable.

The most important difference in the STM32C0 product line is the removal of the PLL.  As coded the default clock source is HSI.

fixes #19210

### CODE REVIEW

**Review by benpicco — APPROVED** [review]
Code looks good to me!
Nice addition.

**Comment by benpicco** [comment]
One more thing: Since the STM32C031C6 has very little memory (32k ROM, 12k RAM) I expect that some applications won't fit on it.

CI tries to build every application for every board, so you need to tell it which ones to exclude for a board because they won't fit. You can do this by running

    dist/tools/insufficient_memory/update_insufficient_memory_board.sh nucleo-c031c6

This will build all applications & tests for `nucleo-c031c6` and update their `Makefile.ci` if the application does not fit.

You then need to commit the updated `Makefile.ci` files.

File: boards/nucleo-c031c6/Makefile.features [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CPU = stm32
+CPU_MODEL = stm32c031c6
+
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_adc
+FEATURES_PROVIDED += periph_i2c
```
Comment (benpicco) [inline boards/nucleo-c031c6/Makefile.features:6]: You'll also need

```
--- a/cpu/stm32/periph/Makefile
+++ b/cpu/stm32/periph/Makefile
@@ -2,7 +2,7 @@ MODULE = periph
 
 # Select the specific implementation for `periph_i2c`
 ifneq (,$(filter periph_i2c,$(USEMODULE)))
-  ifneq (,$(filter $(CPU_FAM),f0 f3 f7 g0 g4 l0 l4 l5 u5 wb wl))
+  ifneq (,$(filter $(CPU_FAM),f0 f3 f7 g0 g4 l0 l4 l5 u5 wb wl c0))
     SRC += i2c_1.c
   else ifneq (,$(filter $(CPU_FAM),f1 f2 f4 l1))
     SRC += i2c_2.c
```

File: cpu/stm32/Makefile.cmsis [hunk 2]
```diff
@@ -3,6 +3,8 @@ PKG_NAME=stm32cmsis
 # The package URL depends on the family of the target STM32
 PKG_URL=https://github.com/STMicroelectronics/cmsis_device_$(CPU_FAM)
 
+# v1.1.0
+PKG_VERSION_c0=7e32bf9d8117ee4c8f6a1d138b814fc24bf4c906
```
Comment (benpicco) [inline cpu/stm32/Makefile.cmsis:7]: You'll also have to remove the `ErrorStatus` enum as that conflicts with some packages, see `cpu/stm32/include/vendor/patches`

Comment (jparker324) [reply cpu/stm32/Makefile.cmsis:7]: I removed the `ErrorStatus` with a local commit in the `build/stm32/cmsis/c0` and then turned that commit into a patch.  It looks like that's how these vendor patches were done.  Let me know if that's not right.

File: boards/nucleo-c031c6/Makefile.features (hunk 1 above)
Comment (jparker324) [reply boards/nucleo-c031c6/Makefile.features:6]: I found this one too.  fixed.

**Comment by benpicco** [comment]
Looks like there is no low power timer on STM32C0 - so you might have to remove `periph_rtt`.
Or is there a timer that keeps running in Deep Sleep?

**Comment by jparker324** [comment]
There is not a dedicated low-power timer `LPTIM` peripheral like on previous ST parts (example [`STM32G071`](https://www.st.com/resource/en/datasheet/stm32g071rb.pdf) section 3.18.4).  Though they offer, according to datasheet [`STM32C031x4/x6`](https://www.st.com/resource/en/datasheet/stm32c031c6.pdf) section 3.15 and 3.16, the LSI as a low-power RC clock source and there are 2 low-power timers that support a couple low-power RTC interrupts.  I don't know if that meets all the requirements to support RIOT's `periph_rtt`.

**Comment by benpicco** [comment]
The only requirement for `periph_rtt` is that it keeps ticking in the deepest sleep state.
But you can also provide that as a follow-up PR and just drop advertising the feature for now.

**Comment by jparker324** [comment]
Alright.  I'll do that.  I still get build errors for `tests/pkg/lvgl`, `tests/pkg/lvgl_touch`, `tests/sys/progress_bar`, and `tests/sys/xtimer_hang`.  `lvgl` and `lvgl_touch` also fail when I try them under `nucleo_g071rb' for example, so that's either expected or local to my environment.  The `progress_bar` and `xtimer_hang` though are "OK" for that board.

**Comment by benpicco** [comment]
What build errors are you getting?

**Comment by jparker324** [comment]
```
tests/pkg/lvgl                          build failed
make: Entering directory '/home/parkerj/RIOT/tests/pkg/lvgl'
Launching build container using image "docker.io/riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/America/New_York:/etc/localtime:ro' -v '/home/parkerj/RIOT:/data/riotbuild/riotbase:delegated' -v '/home/parkerj/.cargo/registry:/data/riotbuild/.cargo/registry:delegated' -v '/home/parkerj/.cargo/git:/data/riotbuild/.cargo/git:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'      -e 'BOARD=nucleo-c031c6' -e 'RIOT_CI_BUILD=1' -e 'DISABLE_MODULE=test_utils_interactive_sync' -e 'DEFAULT_MODULE=' -e 'FEATURES_REQUIRED=' -e 'FEATURES_BLACKLIST=' -e 'FEATURES_OPTIONAL=' -e 'USEMODULE=lvgl_contrib lvgl_extra_layout_flex lvgl_extra_theme_default lvgl_extra_theme_default_dark lvgl_extra_widget_chart lvgl_extra_widget_win' -e 'USEPKG=lvgl'  -w '/data/riotbuild/riotbase/tests/pkg/lvgl/' 'docker.io/riot/riotbuild:latest' make 'BOARD=nucleo-c031c6'   -j12  
Building application "tests_lvgl" for "nucleo-c031c6" with MCU "stm32".

/tmp/ccXrsj5w.s: Assembler messages:
/tmp/ccXrsj5w.s: Error: unaligned opcodes detected in executable segment
make[2]: *** [/data/riotbuild/riotbase/Makefile.base:146: /data/riotbuild/riotbase/tests/pkg/lvgl/bin/nucleo-c031c6/lvgl_core/lv_event.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:76: lvgl_core] Error 2
make: *** [/data/riotbuild/riotbase/Makefile.include:826: pkg-build] Error 2
make: *** [/home/parkerj/RIOT/makefiles/docker.inc.mk:351: ..in-docker-container] Error 2
make: Leaving directory '/home/parkerj/RIOT/tests/pkg/lvgl'
```

```
tests/pkg/lvgl_touch                    build failed
make: Entering directory '/home/parkerj/RIOT/tests/pkg/lvgl_touch'
Launching build container using image "docker.io/riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/America/New_York:/etc/localtime:ro' -v '/home/parkerj/RIOT:/data/riotbuild/riotbase:delegated' -v '/home/parkerj/.cargo/registry:/data/riotbuild/.cargo/registry:delegated' -v '/home/parkerj/.cargo/git:/data/riotbuild/.cargo/git:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'      -e 'BOARD=nucleo-c031c6' -e 'RIOT_CI_BUILD=1' -e 'DISABLE_MODULE=test_utils_interactive_sync' -e 'DEFAULT_MODULE=' -e 'FEATURES_REQUIRED=' -e 'FEATURES_BLACKLIST=' -e 'FEATURES_OPTIONAL=' -e 'USEMODULE=lvgl_contrib lvgl_contrib_touch lvgl_extra_theme_default lvgl_widget_btn lvgl_widget_label random' -e 'USEPKG=lvgl'  -w '/data/riotbuild/riotbase/tests/pkg/lvgl_touch/' 'docker.io/riot/riotbuild:latest' make 'BOARD=nucleo-c031c6'   -j12  
Building application "tests_lvgl_touch" for "nucleo-c031c6" with MCU "stm32".

/tmp/cckmncKC.s: Assembler messages:
/tmp/cckmncKC.s: Error: unaligned opcodes detected in executable segment
make[2]: *** [/data/riotbuild/riotbase/Makefile.base:146: /data/riotbuild/riotbase/tests/pkg/lvgl_touch/bin/nucleo-c031c6/lvgl_core/lv_event.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:76: lvgl_core] Error 2
make: *** [/data/riotbuild/riotbase/Makefile.include:826: pkg-build] Error 2
make: *** [/home/parkerj/RIOT/makefiles/docker.inc.mk:351: ..in-docker-container] Error 2
make: Leaving directory '/home/parkerj/RIOT/tests/pkg/lvgl_touch'
```

```
tests/sys/progress_bar                  build failed
make: Entering directory '/home/parkerj/RIOT/tests/sys/progress_bar'
Launching build container using image "docker.io/riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/America/New_York:/etc/localtime:ro' -v '/home/parkerj/RIOT:/data/riotbuild/riotbase:delegated' -v '/home/parkerj/.cargo/registry:/data/riotbuild/.cargo/registry:delegated' -v '/home/parkerj/.cargo/git:/data/riotbuild/.cargo/git:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'      -e 'BOARD=nucleo-c031c6' -e 'RIOT_CI_BUILD=1' -e 'DISABLE_MODULE=' -e 'DEFAULT_MODULE=test_utils_interactive_sync test_utils_print_stack_usage' -e 'FEATURES_REQUIRED=' -e 'FEATURES_BLACKLIST=' -e 'FEATURES_OPTIONAL=' -e 'USEMODULE=progress_bar xtimer' -e 'USEPKG='  -w '/data/riotbuild/riotbase/tests/sys/progress_bar/' 'docker.io/riot/riotbuild:latest' make 'BOARD=nucleo-c031c6'   -j12  
Building application "tests_progress_bar" for "nucleo-c031c6" with MCU "stm32".

In file included from /data/riotbuild/riotbase/sys/include/auto_init_utils.h:35,
                 from /data/riotbuild/riotbase/sys/auto_init/auto_init.c:26:
/data/riotbuild/riotbase/tests/sys/progress_bar/bin/nucleo-c031c6/preprocessor/preprocessor_successor.h:2: error: unterminated #ifndef
    2 | #ifndef RIOT_PP_SUCCESSOR_H
      | 
make[3]: *** [/data/riotbuild/riotbase/Makefile.base:146: /data/riotbuild/riotbase/tests/sys/progress_bar/bin/nucleo-c031c6/auto_init/auto_init.o] Error 1
make[2]: *** [/data/riotbuild/riotbase/Makefile.base:31: ALL--/data/riotbuild/riotbase/sys/auto_init] Error 2
make[1]: *** [/data/riotbuild/riotbase/Makefile.base:31: ALL--/data/riotbuild/riotbase/sys] Error 2
make: *** [/data/riotbuild/riotbase/Makefile.include:760: application_tests_progress_bar.module] Error 2
make: *** [/home/parkerj/RIOT/makefiles/docker.inc.mk:351: ..in-docker-container] Error 2
make: Leaving directory '/home/parkerj/RIOT/tests/sys/progress_bar'
```

```
tests/sys/xtimer_hang                   build failed
make: Entering directory '/home/parkerj/RIOT/tests/sys/xtimer_hang'
Launching build container using image "docker.io/riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/America/New_York:/etc/localtime:ro' -v '/home/parkerj/RIOT:/data/riotbuild/riotbase:delegated' -v '/home/parkerj/.cargo/registry:/data/riotbuild/.cargo/registry:delegated' -v '/home/parkerj/.cargo/git:/data/riotbuild/.cargo/git:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'      -e 'BOARD=nucleo-c031c6' -e 'RIOT_CI_BUILD=1' -e 'DISABLE_MODULE=core_msg' -e 'DEFAULT_MODULE=test_utils_interactive_sync test_utils_print_stack_usage' -e 'FEATURES_REQUIRED=' -e 'FEATURES_BLACKLIST=' -e 'FEATURES_OPTIONAL=' -e 'USEMODULE=xtimer' -e 'USEPKG='  -w '/data/riotbuild/riotbase/tests/sys/xtimer_hang/' 'docker.io/riot/riotbuild:latest' make 'BOARD=nucleo-c031c6'   -j12  
Building application "tests_xtimer_hang" for "nucleo-c031c6" with MCU "stm32".

/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x45d5b vs 0x333f8)
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x45d5b vs 0x333f8)
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x45d5b vs 0x333f8)
...
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x45d5b vs 0x333f8)
/data/riotbuild/riotbase/tests/sys/xtimer_hang/bin/nucleo-c031c6/cpu/cpu_init.o:/data/riotbuild/riotbase/cpu/stm32/cpu_init.c:359: undefined reference to `cortexm_init'
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x465ed vs 0x34ec4)
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x465ed vs 0x34ec4)
...
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x465ed vs 0x34ec4)
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: DWARF error: section .debug_str is larger than its filesize! (0x465ed vs 0x34ec4)
/data/riotbuild/riotbase/tests/sys/xtimer_hang/bin/nucleo-c031c6/stdio_uart/stdio_uart.o:/data/riotbuild/riotbase/sys/stdio_uart/stdio_uart.c:47: undefined reference to `isrpipe_write_one'
/opt/gcc-arm-none-eabi-10.3-2021.10/bin/../lib/gcc/arm-none-eabi/10.3.1/../../../../arm-none-eabi/bin/ld: /data/riotbuild/riotbase/tests/sys/xtimer_hang/bin/nucleo-c031c6/stdio_uart/stdio_uart.o:/data/riotbuild/riotbase/sys/stdio_uart/stdio_uart.c:73: undefined reference to `isrpipe_read'
collect2: error: ld returned 1 exit status
make: *** [/data/riotbuild/riotbase/Makefile.include:751: /data/riotbuild/riotbase/tests/sys/xtimer_hang/bin/nucleo-c031c6/tests_xtimer_hang.elf] Error 1
make: *** [/home/parkerj/RIOT/makefiles/docker.inc.mk:351: ..in-docker-container] Error 2
make: Leaving directory '/home/parkerj/RIOT/tests/sys/xtimer_hang'
```

**Comment by jparker324** [comment]
I think the failures of `tests/sys/progress_bar` and `tests/sys/xtimer_hang` were due to previous interrupted or cancelled builds.  I cleaned their `build` and `bin` dirs, and they both appear to build fine now.

```tests/sys/progress_bar                  OK```
```tests/sys/xtimer_hang                   OK```

LVGL still fails though.  I also get a brief message on spi_dma before it's `not supported` output, although it's not explicitly `build failed`.

```
tests/periph/spi_dma                    dist/tools/insufficient_memory/../../../tests/periph/spi_dma/Makefile.ci:3: ../spi/Makefile.ci: No such file or directory
make: *** No rule to make target '../spi/Makefile.ci'.  Stop.
not supported
```

**Comment by benpicco** [comment]
> I also get a brief message on `spi_dma` before it's not supported output, although it's not explicitly build failed.

That's because you don't advertise the `periph_dma` feature, but as you also didn't configure any DMA channels for the SPI, that's correct.

File: tests/sys/psa_crypto_eddsa/Makefile.ci [hunk 3]
```diff
@@ -1,14 +0,0 @@
...
-    arduino-uno \
-    atmega328p \
-    atmega328p-xplained-mini \
-    atmega8 \
-    nucleo-f031k6 \
-    nucleo-l011k4 \
-    samd10-xmini \
-    stk3200 \
-    stm32f030f4-demo \
-    #
```
Comment (benpicco) [inline tests/sys/psa_crypto_eddsa/Makefile.ci:14]: You don't want to remove those files

File: examples/javascript/Makefile.ci [hunk 4]
```diff
@@ -34,7 +35,6 @@ BOARD_INSUFFICIENT_MEMORY := \
     nucleo-f334r8 \
     nucleo-f410rb \
     nucleo-g070rb \
-    nucleo-g071rb \
```
Comment (jparker324) [inline examples/javascript/Makefile.ci:37]: undo

**Comment by jparker324** [comment]
I've restored and re-built the `Makefile.ci` files again.  They look better now.  If I had to guess, it was probably another result of earlier cancelled builds.

**Comment by benpicco** [comment]
looks like (at least) `tests/pkg/lvgl_touch` also needs a `Makefile.ci` update

**Comment by jparker324** [comment]
1.  Does anything need to be done about the `tests/sys/ztimer_overhead` error yesterday?

```
make: Entering directory '/tmp/dwq.0.29611624904775513/5b84c211eba9dea6bec9bf777727be8b/tests/sys/ztimer_overhead'
r
/tmp/dwq.0.29611624904775513/5b84c211eba9dea6bec9bf777727be8b/build/tests_ztimer_overhead.elf  tap0 
RIOT native interrupts/signals initialized.
RIOT native board initialized.
RIOT native hardware initialization complete.

Help: Press s to start test, r to print it is ready
READY
s
START
main(): This is RIOT! (Version: buildtest)
ZTIMER_USEC auto_adjust params:
    ZTIMER_USEC->adjust_set = 11
    ZTIMER_USEC->adjust_sleep = 5
ZTIMER_USEC auto_adjust params cleared
zitmer_overhead_set...
min=7 max=43 avg_diff=13
zitmer_overhead_sleep...
min=11 max=255 avg_diff=36
ZTIMER_USEC adjust params for native:
    CONFIG_ZTIMER_USEC_ADJUST_SET    7
    CONFIG_ZTIMER_USEC_ADJUST_SLEEP  11

make: Leaving directory '/tmp/dwq.0.29611624904775513/5b84c211eba9dea6bec9bf777727be8b/tests/sys/ztimer_overhead'
Traceback (most recent call last):
  File "/tmp/dwq.0.29611624904775513/5b84c211eba9dea6bec9bf777727be8b/tests/sys/ztimer_overhead/tests/01-run.py", line 33, in <module>
    sys.exit(run(testfunc))
  File "/tmp/dwq.0.29611624904775513/5b84c211eba9dea6bec9bf777727be8b/dist/pythonlibs/testrunner/__init__.py", line 30, in run
    testfunc(child)
  File "/tmp/dwq.0.29611624904775513/5b84c211eba9dea6bec9bf777727be8b/tests/sys/ztimer_overhead/tests/01-run.py", line 29, in testfunc
    assert auto_adjust_sleep >= adjust_sleep - ADJUST_SLEEP_MARGIN
AssertionError
make: *** [/tmp/dwq.0.29611624904775513/5b84c211eba9dea6bec9bf777727be8b/makefiles/tests/tests.inc.mk:26: test] Error 1
```

2. When running `./dist/tools/insufficient_memory/update_insufficient_memory_board.sh nucleo-c031c6` do you not see the error in the `lvgl` and `lvgl_touch` packages?  How did it/I miss the other `Makefile.ci` you corrected?

```
tests/pkg/lvgl                          build failed
make: Entering directory '/home/parkerj/RIOT/tests/pkg/lvgl'
Launching build container using image "docker.io/riot/riotbuild:latest".
docker run --rm --tty --user $(id -u) -v '/usr/share/zoneinfo/America/New_York:/etc/localtime:ro' -v '/home/parkerj/RIOT:/data/riotbuild/riotbase:delegated' -v '/home/parkerj/.cargo/registry:/data/riotbuild/.cargo/registry:delegated' -v '/home/parkerj/.cargo/git:/data/riotbuild/.cargo/git:delegated' -e 'RIOTBASE=/data/riotbuild/riotbase' -e 'CCACHE_BASEDIR=/data/riotbuild/riotbase' -e 'BUILD_DIR=/data/riotbuild/riotbase/build' -e 'RIOTPROJECT=/data/riotbuild/riotbase' -e 'RIOTCPU=/data/riotbuild/riotbase/cpu' -e 'RIOTBOARD=/data/riotbuild/riotbase/boards' -e 'RIOTMAKE=/data/riotbuild/riotbase/makefiles'      -e 'BOARD=nucleo-c031c6' -e 'RIOT_CI_BUILD=1' -e 'DISABLE_MODULE=test_utils_interactive_sync' -e 'DEFAULT_MODULE=' -e 'FEATURES_REQUIRED=' -e 'FEATURES_BLACKLIST=' -e 'FEATURES_OPTIONAL=' -e 'USEMODULE=lvgl_contrib lvgl_extra_layout_flex lvgl_extra_theme_default lvgl_extra_theme_default_dark lvgl_extra_widget_chart lvgl_extra_widget_win' -e 'USEPKG=lvgl'  -w '/data/riotbuild/riotbase/tests/pkg/lvgl/' 'docker.io/riot/riotbuild:latest' make 'BOARD=nucleo-c031c6'   -j12  
Building application "tests_lvgl" for "nucleo-c031c6" with MCU "stm32".

/tmp/ccXrsj5w.s: Assembler messages:
/tmp/ccXrsj5w.s: Error: unaligned opcodes detected in executable segment
```

**Comment by benpicco** [comment]
That was a great first PR!
Let's continue with GD32E23x :smiley:

**Pull request merged**